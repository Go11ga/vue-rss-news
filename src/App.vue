<template>
    <div id="app">
        
        <div class="preloader" v-if="getLoadingNewsProcess">

            <img src="@/assets/style/images/preloader.gif">
            
        </div>

        <div class="container" v-else>

            <RSSSprite />

            <RSSHeader />

            <RSSControls />

            <router-view></router-view>

        </div>

    </div>
</template>

<script>
import RSSSprite from '@/components/RSSSprite'
import RSSHeader from '@/components/RSSHeader'
import RSSControls from '@/components/RSSControls'
import { mapGetters, mapActions } from 'vuex'
import { POSITION } from "vue-toastification"

export default {
    name: 'RSSApp',
    
    components: {
        RSSSprite,
        RSSHeader,
        RSSControls
    },

    computed: {
        ...mapGetters([
            'getLoadingNewsProcess',
            'getError'
        ])
    },

    methods: {
        ...mapActions([
            'fetchNews'
        ])
    },

    watch: {
        getError(error) {
            this.$toast.error(error.msg, {
                timeout: 2000,
                position: POSITION.BOTTOM_CENTER
            })
        }
    },

    created() {
        this.fetchNews()
    }
}
</script>

<style lang="scss">
@import '@/assets/style/scss/layout.scss';
</style>
