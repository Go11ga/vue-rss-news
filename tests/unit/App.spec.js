import { shallowMount, createLocalVue } from '@vue/test-utils'
import App from '@/App'
import Vuex from 'vuex'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('App.vue данные загрузились', () => {
    let wrapper
    let getters
    let actions
    let store

    beforeEach(() => {
        getters = {
            getLoadingNewsProcess: () => false,
            getError: () => {}
        }

        actions = {
            fetchNews: jest.fn()
        }

        store = new Vuex.Store({
            getters,
            actions
        })

        wrapper = shallowMount(App, {
            localVue,
            store,
            stubs: ['router-view']
        })
    })

    afterEach(() => {
        wrapper.destroy()
    })

    it('проверка имени компонента', () => {
        expect(wrapper.vm.$options.name).toMatch('RSSApp')
    })

    it('snapshot', () => {
        expect(wrapper.element).toMatchSnapshot()
    })

    it('рендеринг при getter getLoadingNewsProcess false', async () => {
        expect(wrapper.html().includes('preloader')).toBe(false)
        expect(wrapper.html().includes('container')).toBe(true)
    })
    
    it('хук created', () => {
        expect(typeof App.created).toBe('function')

        expect(actions.fetchNews).toHaveBeenCalled()
        expect(actions.fetchNews).toHaveBeenCalledTimes(1)
    })
})

describe('App.vue данные не загрузились', () => {
    let wrapper
    let getters
    let actions
    let store

    beforeEach(() => {
        getters = {
            getLoadingNewsProcess: () => true,
            getError: () => {}
        }

        actions = {
            fetchNews: jest.fn()
        }

        store = new Vuex.Store({
            getters,
            actions
        })

        wrapper = shallowMount(App, {
            localVue,
            store,
            stubs: ['router-view']
        })
    })

    afterEach(() => {
        wrapper.destroy()
    })

    it('рендеринг getter getLoadingNewsProcess true', async () => {
        expect(wrapper.html().includes('preloader')).toBe(true)
        expect(wrapper.html().includes('container')).toBe(false)
    })
})
