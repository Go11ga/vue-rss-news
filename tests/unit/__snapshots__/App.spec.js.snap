// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue данные загрузились snapshot 1`] = `
<div
  id="app"
>
  <div
    class="container"
  >
    <rsssprite-stub />
     
    <rssheader-stub />
     
    <rsscontrols-stub />
     
    <router-view-stub />
  </div>
</div>
`;
